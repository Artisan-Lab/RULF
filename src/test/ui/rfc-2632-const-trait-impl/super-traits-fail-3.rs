#![feature(const_trait_impl)]

// revisions: yy yn ny nn
//[yy] check-pass

#[cfg_attr(any(yy, yn), const_trait)]
trait Foo {
    fn a(&self);
}

#[cfg_attr(any(yy, ny), const_trait)]
trait Bar: ~const Foo {}
//[ny,nn]~^ ERROR: ~const can only be applied to `#[const_trait]`

const fn foo<T: ~const Bar>(x: &T) {
    //[yn,nn]~^ ERROR: ~const can only be applied to `#[const_trait]`
    x.a();
}

fn main() {}
