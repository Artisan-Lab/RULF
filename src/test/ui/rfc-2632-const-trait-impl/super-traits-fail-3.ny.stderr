error: ~const can only be applied to `#[const_trait]` traits
  --> $DIR/super-traits-fail-3.rs:12:19
   |
LL | trait Bar: ~const Foo {}
   |                   ^^^

error: aborting due to previous error

